mysite/w6/beanstalk.py

```diff
@@ -1,140 +1,146 @@
 
+    #Also include the amounts equipped on characters
     for toon in session_data.account.safe_characters:
         for food in toon.equipment.foods:
-            if food.codename not in beanstalk_status:
-                continue
-
-            gold_foods[food.codename] += food.amount
-
-    foods_to_10k = [
-        k for k, v in beanstalk_status.items() if v < TEN_K and gold_foods[k] < 10**4
-    ]
-    foods_to_deposit_10k = [
-        k for k, v in beanstalk_status.items() if v < TEN_K and k not in foods_to_10k
-    ]
-
-    foods_to_100k = [
-        k for k, v in beanstalk_status.items() if v < HUNNIT_K and gold_foods[k] < 10**5
-    ]
-    foods_to_deposit_100k = [
-        k for k, v in beanstalk_status.items() if v < HUNNIT_K and k not in foods_to_100k
-    ]
-
-    foods_to_deposit = foods_to_deposit_10k + foods_to_deposit_100k * upgrade_bought
-
-    foods_finished = sum(beanstalk_status.values())
+            if food.codename in gold_foods:
+                gold_foods[food.codename] += food.amount
+
+    foods_ready_to_deposit = []  #Food ready to deposit, including a tag in the name for 10k or 100k
+    foods_to_beanstack = []  #If food is not already Beanstacked and player has less than required amount (10k)
+    foods_to_deposit_for_beanstack = []  #If food is not already Beanstacked and player has enough to do so
+    for foodName, foodValuesDict in session_data.account.sneaking['Beanstalk'].items():
+        if not foodValuesDict['Beanstacked']:
+            if gold_foods[foodName] < BEANSTACK_GOAL:
+                foods_to_beanstack.append(foodName)
+            else:
+                foods_ready_to_deposit.append(f"{getItemDisplayName(foodName)}: 10k Beanstack")
+                foods_to_deposit_for_beanstack.append(foodName)
+
+    super_beanstack_progress: dict[str, int] = dict()
+    for gold_food_code in gold_foods:
+        total_owned = gold_foods[gold_food_code]
+
+        # remove 10k from the amount of gold food owned if the first 10k hasn't been deposited yet
+        adjusted_total_owned = total_owned if session_data.account.sneaking['Beanstalk'][gold_food_code]['Beanstacked'] else total_owned - BEANSTACK_GOAL
+
+        # mark progress as 0% if less than 10k is owned
+        super_beanstack_progress[gold_food_code] = max(0, adjusted_total_owned)
+
+    foods_to_super_beanstack = []  # If food is not already Beanstacked and player has less than required amount (10k)
+    for foodName, foodValuesDict in session_data.account.sneaking['Beanstalk'].items():
+        if not foodValuesDict['SuperBeanstacked']:
+            if super_beanstack_progress[foodName] < SUPER_BEANSTACK_GOAL:
+                foods_to_super_beanstack.append(foodName)
+            else:
+                foods_ready_to_deposit.append(f"{getItemDisplayName(foodName)}: 100k Super Beanstack")
+
+    foods_finished = sum([v['Beanstacked'] for v in session_data.account.sneaking['Beanstalk'].values()])
```

Comment:
Forgot to include the number of `SuperBeanstacked` here. Your screenshot on Discord says "12/30" now instead of "16/30"

Reply:
Good catch, thank you. Adding that now.

Reply:
![image](https://github.com/TwoSpookyBoos/IdleOnAutoReviewBot/assets/150628389/dfd9f25f-fb89-4204-aaee-e201c5f79aa7)